libminifi/include/utils/MathUtils.h

```diff
@@ -0,0 +1,43 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <math.h>
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+
+class MathUtils {
+ public:
+  static double round_to(double original, int8_t precision) {
+    if (precision < 0) {
+      return original;
+    } else {
+      double power_ten = pow(10, precision);
+      return std::round(original * power_ten) / power_ten;
```

Comment:
as far as I know, precision has a different meaning, it is the number of "significant digits", so pinting `0.00123` with a precision of `2` should result in `0.0012` instead of `0` as this function does

Reply:
a thought: since the `rapidjson::Document::Accept` is templated, we could wrap a writer (`rapidjons::Writer<...>`, `rapidjson::PrettyWriter<...>`) in a custom class, inheriting from it and shadowing the `Double` method to implement our custom double formatting, although this might not be trivial

Reply:
You are right, changed the function to round_to_decimal_places so its more clear what the function does.